### PR TITLE
feat(api/vault): Add endpoints for vault balance retrieval and update

### PR DESCRIPTION
**Pull request by Jagadeeshftw**

### CODE REVIEW

**Comment by Jagadeeshftw** [comment]
@djeck1432 Can you review this PR?

File: web_app/api/vault.py [hunk 1]
```diff
@@ -3,41 +3,89 @@
...
             amount=request.amount,
-            symbol=request.symbol
+            symbol=request.symbol,
         )
     except (ValueError, TypeError) as e:
         logger.error(f"Invalid input data: {str(e)}")
         raise HTTPException(status_code=400, detail=str(e))
+
+
+@router.get("/balance", response_model=VaultBalanceResponse)
```
Comment (djeck1432) [inline web_app/api/vault.py:53]: Please, add `api` -> "/api/balance"

File: web_app/api/vault.py [hunk 2]
```diff
@@ -3,41 +3,89 @@
...
+    """
+    balance = deposit_connector.get_vault_balance(wallet_id=wallet_id, symbol=symbol)
+    if balance is None:
+        raise HTTPException(
+            status_code=404, detail="Vault not found or user does not exist"
+        )
+    return VaultBalanceResponse(wallet_id=wallet_id, symbol=symbol, amount=balance)
+
+
+@router.post("/add_balance", response_model=UpdateVaultBalanceResponse)
```
Comment (djeck1432) [inline web_app/api/vault.py:70]: Please, add `api` -> "/api/add_balance"

File: web_app/api/vault.py [hunk 3]
```diff
@@ -3,41 +3,89 @@
...
+        deposit_connector.add_vault_balance(
+            wallet_id=request.wallet_id, symbol=request.symbol, amount=request.amount
+        )
+        return UpdateVaultBalanceResponse(
+            wallet_id=request.wallet_id,
+            symbol=request.symbol,
+            amount=request.amount,
+            message="Vault balance updated successfully",
+        )
+    except Exception as e:
```
Comment (djeck1432) [inline web_app/api/vault.py:88]: which particular exception do you want to catch, avoid using generic Exception class

File: web_app/api/vault.py [hunk 4]
```diff
@@ -3,41 +3,89 @@
...
+    """
+    try:
+        deposit_connector.add_vault_balance(
+            wallet_id=request.wallet_id, symbol=request.symbol, amount=request.amount
+        )
+        return UpdateVaultBalanceResponse(
+            wallet_id=request.wallet_id,
+            symbol=request.symbol,
+            amount=request.amount,
+            message="Vault balance updated successfully",
```
Comment (djeck1432) [inline web_app/api/vault.py:86]: This field is not necessary, please remove it [message]

File: web_app/api/vault.py [hunk 5]
```diff
@@ -3,41 +3,89 @@
...
+    Add balance to a user's vault for a specific token.
+    """
+    try:
+        deposit_connector.add_vault_balance(
+            wallet_id=request.wallet_id, symbol=request.symbol, amount=request.amount
+        )
+        return UpdateVaultBalanceResponse(
+            wallet_id=request.wallet_id,
+            symbol=request.symbol,
+            amount=request.amount,
```
Comment (djeck1432) [inline web_app/api/vault.py:85]: this is incorrect, amount=request.amount + vault_amount

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Comment by djeck1432** [comment]
@Jagadeeshftw any updates?

File: web_app/api/vault.py (hunk 5 above)
Comment (Jagadeeshftw) [reply web_app/api/vault.py:85]: Here `vault_amount` seems not defined, from where we need to get it? @djeck1432

File: web_app/api/vault.py (hunk 1 above)
Comment (Jagadeeshftw) [reply web_app/api/vault.py]: Updated!

File: web_app/api/vault.py (hunk 2 above)
Comment (Jagadeeshftw) [reply web_app/api/vault.py]: Updated!

File: web_app/api/vault.py (hunk 3 above)
Comment (Jagadeeshftw) [reply web_app/api/vault.py]: Updated!

File: web_app/api/vault.py (hunk 4 above)
Comment (Jagadeeshftw) [reply web_app/api/vault.py]: Updated!

File: web_app/api/vault.py [hunk 6]
```diff
@@ -3,41 +3,88 @@
...
+    Add balance to a user's vault for a specific token.
+    """
+    try:
+        deposit_connector.add_vault_balance(
+            wallet_id=request.wallet_id, symbol=request.symbol, amount=request.amount
+        )
+        return UpdateVaultBalanceResponse(
+            wallet_id=request.wallet_id,
+            symbol=request.symbol,
+            amount=request.amount + vault_amount
```
Comment (djeck1432) [inline web_app/api/vault.py:85]: I think you didn't test it, so vault_amount wasn't predefined and this code will raise error
You should return `deposit_connector.add_vault_balance` instance and fetch it from there then:
```
vault = deposit_connector.add_vault_balance(
            wallet_id=request.wallet_id, symbol=request.symbol, amount=request.amount
        )
  ....
  amount=vault.amount
```

make sure that ` deposit_connector.add_vault_balance` will return vault instance if not, adjust it

**Review by djeck1432 — CHANGES_REQUESTED** [review]

File: web_app/api/vault.py (hunk 6 above)
Comment (Jagadeeshftw) [reply web_app/api/vault.py]: Updated!   `deposit_connector.add_vault_balance` will return `None` hence, it was adjusted.

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Pull request merged**